src/app/AttributeCache.cpp

```diff
@@ -77,6 +88,18 @@ void AttributeCache::OnReportEnd()
 {
     std::set<std::tuple<EndpointId, ClusterId>> changedClusters;
 
+    for (auto & endpointIter : mCache)
+    {
+        for (auto & clusterIter : endpointIter.second)
+        {
+            if (clusterIter.second.mPendingDataVersion.HasValue())
+            {
+                clusterIter.second.mCommittedDataVersion = clusterIter.second.mPendingDataVersion;
```

Comment:
So this does not commit the data version until the report ends entirely.... I thought we had talked about committing it once we moved on from a given cluster to a different cluster.  Does that not work for some reason?  It does require tracking what the last path we saw was, of course.

Reply:
for read attribute paths,  we may have C1A1, C2A2, C1A2, things related to same cluster may not be put together, now, we don't have mechanism to gather them together. Then in report engine side, https://github.com/project-chip/connectedhomeip/blob/master/src/app/reporting/Engine.cpp#L129, the generated attributes for same cluster may not in together, so we don't know when we are moving from one cluster to different cluster.

Reply:
Just look at Jerry's below comment, yes, agree, we should go for "committing it once we moved on from a given cluster to a different cluster", update the code, thanks